src/NHibernate/Linq/Visitors/PreTransformationParameters.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.Collections.Generic;
+using System.Linq.Expressions;
+using NHibernate.Engine;
+using Remotion.Linq.Parsing.ExpressionVisitors.TreeEvaluation;
+
+namespace NHibernate.Linq.Visitors
+{
+	/// <summary>
+	/// Contains the information needed by <see cref="NhRelinqQueryParser.PreTransform"/> to perform an early transformation.
+	/// </summary>
+	public class PreTransformationParameters
```

Comment:
Added a separate class, in order to be able to add addtitional information to pre-transformation phase in the future. One example would be to add support for:
- A inline constant value would not be translated to a sql parameter, [described here](https://github.com/nhibernate/nhibernate-core/pull/2335#issuecomment-602035664).

in order to add such feature we would need to extend `NhLinqExpression` constructor with `NhQueryableOptions` parameter that would be passed to `PreTransform` method, which would contain the information whether the parameters needs to be inlined or not.